src/librustc/ty/maps/config.rs

```diff
@@ -156,6 +156,18 @@ impl<'tcx> QueryDescription<'tcx> for queries::const_eval<'tcx> {
     fn describe(tcx: TyCtxt, key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>) -> String {
         format!("const-evaluating `{}`", tcx.item_path_str(key.value.instance.def.def_id()))
     }
+
+    #[inline]
+    fn cache_on_disk(key: Self::Key) -> bool {
+        key.value.instance.def_id().is_local()
```

Comment:
Can non-local consts be loaded from metadata? Otherwise we just might want to cache everything.

Reply:
not yet (at least not in general, some are), but it is planned

Reply:
Let's just return true here then.